modules/appgw/variables.tf

```diff
@@ -0,0 +1,228 @@
+    - `backend_hostname`
+      - example value: "host.name
+      - override host header with a custom host name, when not set defaults to the host header of the original request, mutually exclusive with `backend_hostname_from_backend` (see above)
+    - `backend_timeout`
+      - example value: 60
+      - timeout for backend's response in seconds, default to 60s
+    - `backend_path`
+      - example value: "/path"
+      - path prefix, in case we need to shift the url path for the backend, optinal, can be omited
+    - `cookie_based_affinity`
+      - example value: "Enabled"
+      - cookie based routing, defaults to `Enabled`
+    - `affinity_cookie_name`
+      - example value: "SomeCookieName"
+      - name of the affinity cookie, defaults to Azure default name
+    - `backend_root_certs`
+      - example value: `{ some_root_cert = "./files/self_signed.crt" }`
+      - (v2 only) a map of custom root certificates used to sign backend's certificate.
+    
+    - `probe_path`
+      - example value: "/healthcheck"
+      - url for the health check endpoint, this property controls if the custom probe is created or not. If this is not set, http settings will have the property `Use custom probe` set to `No`
+    - `probe_host`
+      - example value: "www.example.com"
+      - host header for the health check probe, when omited sets the `Pick host name from backend HTTP settings` to `Yes`. Cannot be omited when `backend_hostname` nor `backend_hostname_from_backend` are not set.
+    - `probe_port`
+      - example value: `80`
+      - (v2 only) port for the health check, defaults to default protocol port
+    - `probe_interval`
+      - example value: 2
+      - probe interval in seconds, defaults to 5
+    - `probe_timeout`
+      - example value: 30
+      - probe timeout in seconds, defaults to 30
+    - `probe_theshold`
+      - example value: 2
+      - number of failed probes until the bakckend is marked as down, defaults to 2
+    - `probe_match_code`
+      - example value: [200]
+      - a list of acceptible http response codes. `probe_match_code` controls the custom match condition for a health probe, if not set, it disables the custom match conditions.
+    - `probe_match_body`
+      - example value: ""
+      - a snippet of the backend response that can be matched for health check conditions, defaults to an empty string
+    
+    - `ssl_certificate_path`
+      - example value: "cert/path"
+      - a path to a certificate in `.pfx` format. Required only for `https` listeners
+    - `ssl_certificate_pass`
+      - example value: "cert_password"
+      - a matching password for the certificate specified in `ssl_certificate_path`
+    ```
+  EOF
+  validation {
```

Comment:
Wow, nice to see validation in use :)

Reply:
setting up the config is to complex to just depend on documentation :)